File: Tutorial5-2Dagger-Android-SubComponents/src/main/java/com/example/tutorial5_2dagger_android_subcomponents/ToastMaker.java
package com.example.tutorial5_2dagger_android_subcomponents;

import android.content.Context;
import android.widget.Toast;

public class ToastMaker {

    private Context context;

    public ToastMaker(Context context) {
        this.context = context;
    }

    public void showToast(String message) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
    }


}

File: Tutorial5-2Dagger-Android-SubComponents/src/main/java/com/example/tutorial5_2dagger_android_subcomponents/model/ToastMaker.kt
package com.example.tutorial5_2dagger_android_subcomponents.model

import android.content.Context
import android.widget.Toast

class ToastMaker(private val context: Context) {

    fun showToast(message: String) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
    }

}